### ISSUE TITLE
Note opens at end of file rather than beginning

### ISSUE THREAD

**Issue by aguador** [tomboy-notes/tomboy-ng#193]
I thought I filed a report with 0.28, but apparently not. This is in normal, not single note mode.

**Comment by davidbannon** [comment]
Err, thats strange.  Yes, I reckon we discussed it.  But, maybe I could not replicate it  ?  I certainly cannot replicate it now, just opened half a dozen notes and they all opened with the cursor just under the heading.

So, are you doing anything particular when you open a note ?  Is it happening every time, just one particular note ?

Davo

**Comment by aguador** [comment]
Every time and on both Mageia and MX Linux -- both Gtk2. I normally open from the systray icon, but just used the search window, and the same thing. The note opens with the cursor at the end.

**Comment by davidbannon** [comment]
that is very strange.  I doubt there is anything about the distro relevant but I will fire up my VMs later today and see.
Davo

**Comment by davidbannon** [comment]
aguador, I still cannot replicate this.  One thing that did occur to me was maybe you are re-opening an open note ?

-ng has a facility built in that is particularly beneficial to 'workspace' users.  I always have four active workspaces, supose you open a note in one workspace, and, perhaps, scroll to the end of it. Then, switch to another workspace. Then click the system tray icon, select that same note. It does not open a new instance of that note, it goes and gets it from the other workspace and brings to to your current one.  And, of course, when it arrives in your current workspace, the cursor is where ever you left it previously.

A similar but more intuitive thing happens if you are within one workspace and the note you want is open but obscured.

Davo

**Comment by aguador** [comment]
Nope, this happens when first opening a note, and it happens in both my Mageia (2 machines) and MX Linux installs. (Note MX Linux install is more a test bed so just has a few short test notes.)

What else in the config or ENV might be affecting this? The only differences I may have from your configs: dark mode, Spanish, no sync set up (except, I think in MX install), for configs: locale. Copy of Mageia config file attached and ENV variables (with a bit of data blacked out).
[Cauldron_ENVs.txt](https://github.com/tomboy-notes/tomboy-ng/files/4908204/Cauldron_ENVs.txt)

[tomboy-ng.cfg.txt](https://github.com/tomboy-notes/tomboy-ng/files/4908193/tomboy-ng.cfg.txt)

**Comment by davidbannon** [comment]
Wow, thats a hard one then.  In the FormShow event, I have code like this -
```
    if SearchedTerm <> '' then
        FindIt(SearchedTerm, True, False)
    else begin
        KMemo1.ExecuteCommand(ecEditorTop);
        KMemo1.ExecuteCommand(ecDown);         
    end;
```
There is nothing you can have in settings or your env that will mess with those lines.  In any case except an active search, we move to top of page and then down one line. 

Maybe I send you a binary with some extra debugging to make sure those lines are being executed ??  

I will really need to see it here. I don't have a MX testbed and my Mageia ones are dated perhaps ??  Will get/update as appropriate. Your Mageia uses Enlightenment ?  Your MX uses ?

Davo

**Comment by davidbannon** [comment]
No, no progress.
* I updated my mageia 7.1 - lots of download.  Tested with Enlightenment, KDE and Cinnamon. 
* Installed 19.2 MXLinux, again, no problems.

In every case,  a note opens with the cursor just under the title.

Hey, are you using the QT5 version or GTK2 ?

Please make a new note, add a couple of lines, close and reopen. Same result ?

Davo

**Comment by aguador** [comment]
Newly created saved and reopened note -- with the cursor at the end of the file. Consistent across the machines: E24.1, E22.4, XFCE 14.4 (the latter is the MX Linux DE). If you can throw together some debugging . . . or will a simple strace work?

![New_test_note_Mga8_20200715](https://user-images.githubusercontent.com/8114758/87578174-5767ac80-c6d4-11ea-95d7-2b7867fe3b23.jpg)

**Comment by aguador** [comment]
Sorry, I should have reduced the size on the jpg. 
Just answered by own question: strace does nothing. 
BTW "--debug-spell" switch is not included in the current man file.

**Comment by davidbannon** [comment]
OK, this really does not make sense.  I'll mock up something but not immediately, I have made a mess of the plan to get rid of TStringGrid from Recover form, will need to back out as I have discovered it really does have to be TStringGrid there, only way I can display the really long error message generated when a note with bad xml is discovered.
Give me a day or so ....
Davo

**Comment by aguador** [comment]
What's the most basic difference between your systems and mine? Language.

If I open ng with --lang=en, the notes open properly with the cursor sitting on the line below the title, in Spanish at the end. Tested with Gtk in both Mageia and MX Linux.

DK why language should affect this, but there is some weird interaction going on . . . . In any event this would be the invisible pink elephant in the room we have been ignoring, and which explains why you cannot reproduce it, and why there are no other complaints about it!

Changing issue . . .

(Aside, of no importance for this or probably anything else: opening ng in terminal no matter the language gets this: "Gtk-Message: 08:53:22.014: Failed to load module "atk-bridge")

**Comment by davidbannon** [comment]
Oh, how did we miss that ?

Yep, does it on my laptop in spanish too.  The block is being skipped because it thinks there is an active search in progress. I should be able to work that out !

Thanks aguador, a language dependency is the last thing I would have expected there !

Davo

**Comment by davidbannon** [comment]
Easy fixed. I had something looking for the word 'Search' in the search term field, but you have translated that to spanish and it no longer matched ! 
That has been like that for a looooong time !
Thanks !
Davo
